group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8686,6 +8695,88 @@ public void testListGroups() {
+        );
+    }
+
+    @Test
+    public void testDescribeGroupStableForStaticMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        context.staticMembersJoinAndRebalance(
+            "group-id",
+            "leader-instance-id",
+            "follower-instance-id"
+        );
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup("group-id", false);
+
+        List<DescribeGroupsResponseData.DescribedGroup> describedGroups =
+            context.describeGroups(Collections.singletonList("group-id"));
+
+        assertEquals("consumer", describedGroups.get(0).protocolType());
+        assertEquals("range", describedGroups.get(0).protocolData());
+        assertEquals(group.allMembers().stream().map(member -> member.memberId()).collect(Collectors.toList()),
+            describedGroups.get(0).members().stream().map(member -> member.memberId()).collect(Collectors.toList())
+        );
+        assertEquals(group.allMembers().stream().map(member -> member.groupInstanceId().get()).collect(Collectors.toList()),
+            describedGroups.get(0).members().stream().map(member -> member.groupInstanceId()).collect(Collectors.toList())
+        );
+    }
+
+    @Test
+    public void testDescribeGroupRebalancing() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        GenericGroup group = context.createGenericGroup("group-id");
+        JoinGroupResponseData responseData = context.joinGenericGroupAsDynamicMemberAndCompleteJoin(
+            new JoinGroupRequestBuilder()
+                .withGroupId("group-id")
+                .withMemberId(UNKNOWN_MEMBER_ID)
+                .withDefaultProtocolTypeAndProtocols()
+                .build()
+        );
+
+        List<DescribeGroupsResponseData.DescribedGroup> describedGroups =
+            context.describeGroups(Collections.singletonList("group-id"));
+
+        assertTrue(group.isInState(COMPLETING_REBALANCE));
+        assertEquals(responseData.protocolType(), describedGroups.get(0).protocolType());
+        assertEquals("", describedGroups.get(0).protocolData());
+        assertTrue(describedGroups.get(0).members().stream().map(member -> member.memberId()).collect(Collectors.toList())
+            .contains(responseData.memberId()));
+        describedGroups.get(0).members().forEach(member -> assertTrue(member.memberMetadata().length == 0));
+        describedGroups.get(0).members().forEach(member -> assertTrue(member.memberAssignment().length == 0));
+    }
+
+    @Test
+    public void testDescribeGroupsGroupIdNotFoundException() {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        context.createSnapshot();
```

Comment:
nit: Instead of doing this, would it make sense to create the first snapshot in the constructor of the test context?